src/Storages/MaterializedView/RefreshTask.cpp

```diff
@@ -263,6 +294,28 @@ void RefreshTask::stop()
     scheduleRefresh(guard);
 }
 
+void RefreshTask::start_replicated()
+{
+    if (!coordination.coordinated)
+        throw Exception(ErrorCodes::INCORRECT_QUERY, "Refreshable materialized view is not coordinated.");
+    const auto zookeeper = view->getContext()->getZooKeeper();
+    String path = coordination.path + "/paused";
+    auto code = zookeeper->tryRemove(path);
+    if (code != Coordination::Error::ZOK && code != Coordination::Error::ZNONODE)
+        throw Coordination::Exception::fromPath(code, path);
+}
+
+void RefreshTask::stop_replicated(const String & reason)
```

Comment:
If I understand correctly, this function is not getting called when we start restoring backup? The only other place in the code where we create the `paused` znode is in the constructor of `RefreshTask`, and it is only run in the constructor of MV storage. So the node shouldn't be created there either unless we detach and create all the tables when restoring backup (I briefly looked at `RestorerFromBackup` and didn't find any such logic there, but maybe I missed something)?

Reply:
> and it is only run in the constructor of MV storage

Yes, so RMVs created by RestorerFromBackup will be paused initially.

If the RMV is not part of the backup (e.g. it's in a different database from the one we're restoring), it won't be paused. That seems ok.

Reply:
So, we want to only pause the RMVs created by the RestorerFromBackup, got it